File: Safari-611.1.10/LayoutTests/http/tests/security/contentSecurityPolicy/resources/report-file-path.php
<?php
require_once '../../../resources/portabilityLayer.php';

if (isset($_GET['test'])) {
    $reportFilePath = sys_get_temp_dir() . "/" . str_replace("/", "-", $_GET['test']) . ".csp-report.txt"; 
} elseif (isset($_SERVER["HTTP_REFERER"]) and strpos($_SERVER["HTTP_REFERER"], '/resources/') === false) {
    $reportFilePath = sys_get_temp_dir() . "/" . str_replace("/", "-", parse_url($_SERVER["HTTP_REFERER"], PHP_URL_PATH)) . ".csp-report.txt"; 
} else {
    header("HTTP/1.1 500 Internal Server Error");
    echo "This script needs to know the name of the test to form a unique temporary file path. It can get one either from HTTP referrer, or from a 'test' parameter.\n";
    exit();
}

?>

File: LayoutTests/http/tests/security/contentSecurityPolicy/resources/report_file_path.py
#!/usr/bin/env python3

import os
import sys
import tempfile
from urllib.parse import parse_qs, urlparse

referer = os.environ.get('HTTP_REFERER', None)
test = parse_qs(os.environ.get('QUERY_STRING', ''), keep_blank_values=True).get('test', [None])[0]

if test is not None:
    report_filepath = os.path.join(tempfile.gettempdir(), '{}.csp-report.txt'.format(test.replace('/', '-')))
elif referer is not None and '/resources/' not in referer:
    report_filepath = os.path.join(tempfile.gettempdir(), '{}.csp-report.txt'.format(urlparse(referer).path.replace('/', '-')))
else:
    sys.stdout.write(
        'status: 500\r\n'
        'Content-Type: text/html\r\n\r\n'
        'This script needs to know the name of the test to form a unique temporary file path. It can get one either from HTTP referrer, or from a \'test\' parameter.\n'
    )
    sys.exit(0)
